### ISSUE TITLE
This resource can only be used in the context of an organization, "" is a user

### ISSUE THREAD

**Issue by MaXeraph** [integrations/terraform-provider-github#830]
### Terraform Version
* Terraform version: `1.0.0`
* Github version: `4.11.0`

### Affected Resource(s)
- github_membership

### Terraform Configuration Files
```hcl
resource "github_membership" "configuration" {
  username = var.username
  role     = var.role
}

variable "username" {}

variable "role" {
  default = "member"
}

provider "github" {
  owner = "foo"
}
```

### Debug Output
Debug/Trace doesn't show any new output that could specify the error.

### Panic Output
N/A

### Expected Behavior
Should be able to output a plan to remove `github_membership` resource belonging to an **organization**.

### Actual Behavior
Errors:
```bash
Error: This resource can only be used in the context of an organization, "" is a user.

  with module.users_foo["bar"].github_membership.configuration,
  on modules/user-memberships/main.tf line 1, in resource "github_membership" "configuration":
   1: resource "github_membership" "configuration" {
```

### Steps to Reproduce
1. `terraform plan --refresh=false`

### Important Factoids
* Errors and crashes when running `plan` on Jenkins. There were no changes on our Jenkins configuration, and we were able to run `plan` and `apply` with no errors prior to this config modification.
* Locally running the same configuration with `terraform plan --refresh=false` successfully create a plan. However, it is run with my own Github token, supplied in the `provider` resource. Whereas Jenkins utilizes a global token (which has been working fine until now) in env `GITHUB_TOKEN`

### References
Possibly related to https://github.com/integrations/terraform-provider-github/issues/501

**Comment by martyca** [comment]
I am experiencing the same behavior when trying to import an existing resource.
# Terraform Version
Terraform version: 1.0.0
Github version: 4.11.0
# Affected Resource(s)
github_team

# Steps to Reproduce
```
terraform import github_team.kb_tf_team.my_team 123456
```

**Comment by roobert** [comment]
Hey, I have also come across this same problem. The issue happens for me when using modules, it seems like the provider `owner` value is not inherited by modules and `GITHUB_OWNER` must be  set in the environment in order to fix.

**Comment by kanekotic** [comment]
is there any work around for this?

**Comment by kanekotic** [comment]
still happening in 4.13.0

@roobert workaround of environment variables still works, but it relies now on a deprecated functionality. 
```
╷
│ Warning: "organization": [DEPRECATED] Use owner (or GITHUB_OWNER) instead of organization (or GITHUB_ORGANIZATION)
│ 
│   with provider["registry.terraform.io/hashicorp/github"],
│   on <empty> line 0:
│   (source code not available)
│ 
```

**Comment by guillermotti** [comment]
Yes, as @hazcod said, with v4.17.0 I faced the same problem creating a new team

**Comment by lcaille** [comment]
Hi, same error with v4.18.0 too.

**Comment by hazcod** [comment]
Same with v4.19.0... Workaround:
```
export GITHUB_OWNER="${TF_VAR_github_owner}"
export GITHUB_TOKEN="${TF_VAR_github_token}"
```

**Comment by mwos-sl** [comment]
exporting `GITHUB_OWNER` does not solve the problem for me. 
v4.24.1

**Comment by hazcod** [comment]
Still needing to do `export GITHUB_TOKEN="${TF_VAR_github_token}"` to workaround.

**Comment by kmdouglass** [comment]
> Same with v4.19.0... Workaround:
> 
> ```
> export GITHUB_OWNER="${TF_VAR_github_owner}"
> export GITHUB_TOKEN="${TF_VAR_github_token}"
> ```

This solution works for me with `integrations/github` provider version 4.28.0 so long as the provider configuration block is empty. In other works:

```hcl
provider "github" {
  token = "var.token"
  owner = "my-org"
}
```

will fail even with the `GITHUB_TOKEN` and `GITHUB_OWNER` environment variables set, but 

```hcl
provider "github" {}
```

will work if the environment variables are set.

**Comment by alexlop29** [comment]
> > Same with v4.19.0... Workaround:
> > ```
> > export GITHUB_OWNER="${TF_VAR_github_owner}"
> > export GITHUB_TOKEN="${TF_VAR_github_token}"
> > ```
> 
> This solution works for me with `integrations/github` provider version 4.28.0 so long as the provider configuration block is empty. In other words:
> 
> ```terraform
> provider "github" {
>   token = "var.token"
>   owner = "my-org"
> }
> ```
> 
> will fail even with the `GITHUB_TOKEN` and `GITHUB_OWNER` environment variables set, but
> 
> ```terraform
> provider "github" {}
> ```
> 
> will work if the environment variables are set.

+1 with v.5.7.0

**Comment by kfcampbell** [comment]
> it seems like the provider owner value is not inherited by modules and GITHUB_OWNER must be set in the environment in order to fix.

I wonder if that's a thing we can adjust in our code or a limitation of Terraform as a whole.

**Comment by sergio91zulu** [comment]
Work For me, Provider configure:

```
variable "GITHUB_TOKEN" {
  description = "Github Token Admin"
  type        = string
}

variable "GITHUB_OWNER" {
  description = "Github Owner Organization"
  type        = string
}
```
```
  required_providers {
    github = {
      source  = "integrations/github"
      version = "~> 4.0"
    }
  }
```
```
provider "github" {
  token     = var.GITHUB_TOKEN
  owner     = var.GITHUB_OWNER
}
```
export TF_VAR_GITHUB_TOKEN={personal_token}
export TF_VAR_GITHUB_OWNER={org-name}
export GITHUB_OWNER="${TF_VAR_GITHUB_OWNER}"
export GITHUB_TOKEN="${TF_VAR_GITHUB_TOKEN}"

**Comment by shaun-rutherford** [comment]
I'm in the process of updating our TF code base from the 2.x version of github when I ran into this issue.

It looks like this issue starts in v3.0.0.
I tested on v2.9.1 and it works without issue.

**Comment by kfcampbell** [comment]
Hmm...[here](https://github.com/integrations/terraform-provider-github/compare/48c2030ff40c2732bdb95cc6b7eae8b4d3d21305...ec1ed2e044f6001899b07d0dbcb172baa14d3a76)'s the diff between those two tags. There's some changes in github/config.go that could potentially be at fault.

**Comment by thetimbecker** [comment]
BEWARE: this error is very misleading. We thought the Go library needed because of https://github.com/integrations/terraform-provider-github/issues/1391

For us, it turns out our Personal Access Token just stopped working for some reason. We made a new token and everything started working again.

Seems like this has come up a lot: https://github.com/integrations/terraform-provider-github/search?q=in+the+context+of+an+organization+is+a+user&type=issues

Maybe this can be added to the README or something as a troubleshooting step?

**Comment by kfcampbell** [comment]
@thetimbecker agreed that usually the cause is an invalid PAT! Would you be interested in opening up a PR to update the docs accordingly?

**Comment by aiham** [comment]
> > > Same with v4.19.0... Workaround:
> > > ```
> > > export GITHUB_OWNER="${TF_VAR_github_owner}"
> > > export GITHUB_TOKEN="${TF_VAR_github_token}"
> > > ```
> > 
> > 
> > This solution works for me with `integrations/github` provider version 4.28.0 so long as the provider configuration block is empty. In other words:
> > ```terraform
> > provider "github" {
> >   token = "var.token"
> >   owner = "my-org"
> > }
> > ```
> > 
> > 
> >     
> >       
> >     
> > 
> >       
> >     
> > 
> >     
> >   
> > will fail even with the `GITHUB_TOKEN` and `GITHUB_OWNER` environment variables set, but
> > ```terraform
> > provider "github" {}
> > ```
> > 
> > 
> >     
> >       
> >     
> > 
> >       
> >     
> > 
> >     
> >   
> > will work if the environment variables are set.
> 
> +1 with v.5.7.0

Here's another confirmation 4 months later. This fixed it for me in v5.18.0.

**Comment by thetimbecker** [comment]
> @thetimbecker agreed that usually the cause is an invalid PAT! Would you be interested in opening up a PR to update the docs accordingly?

Clearly I never got around to this 😅

Unfortunately I can't take the time to do it well right now, especially since the issue it's fresh in my mind anymore.
